.github/workflows/ci.yml

```diff
@@ -4,26 +4,80 @@ on:
   - pull_request
 
 jobs:
-  test:
+  setup_matrix:
+    name: 'Setup Test Matrix'
     runs-on: ubuntu-latest
+    timeout-minutes: 40
+    outputs:
+      beaker_setfiles: ${{ steps.get-outputs.outputs.beaker_setfiles }}
+      puppet_major_versions: ${{ steps.get-outputs.outputs.puppet_major_versions }}
+      puppet_unit_test_matrix: ${{ steps.get-outputs.outputs.puppet_unit_test_matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install the Puppet PDK
+        run: |
+          CODENAME=$(grep UBUNTU_CODENAME /etc/os-release | cut -d '=' -f2)
+          wget https://apt.puppet.com/puppet-tools-release-$CODENAME.deb
+          sudo apt-get update
```

Comment:
This update isn't really needed, is it? dpkg doesn't use it anyway so it just slows you down.